### PR TITLE
feat(commentpanel): #MAG-357 commentpanel+sub cpts

### PR DESCRIPTION
**Pull request by ThomasSevagen**
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/components/comment-confirmation-modal/style.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  borderColor: "#FF8D2E",
+  "&:hover": {
+    borderColor: "#FF8D2E",
+  },
+};
+
+export const deleteButtonStyles: SxProps<Theme> = {
+  fontSize: "1.6rem",
+  borderRadius: "1rem",
+  backgroundColor: "#FF8D2E",
```
Comment (niko-amosse) [inline frontend/src/components/comment-confirmation-modal/style.ts:46]: constantes pour les couleurs ?

File: frontend/src/components/comment-panel-item/style.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      border: `1px solid ${isEditing ? "#2A9AC7" : "#E4E4E4"}`,
+      pointerEvents: "none",
+    },
+
+    "&:hover fieldset": {
+      border: `1px solid ${isEditing ? "#2A9AC7" : "#E4E4E4"}`,
+    },
+
+    "&.Mui-focused fieldset": {
+      border: `1px solid ${isEditing ? "#2A9AC7" : "#E4E4E4"}`,
```
Comment (niko-amosse) [inline frontend/src/components/comment-panel-item/style.ts:78]: const ?

File: frontend/src/components/comment-panel/utils.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  if (!comments?.length) return [];
+
+  const sortedComments = [...comments].sort(
+    (a, b) => dayjs(a.creationDate).valueOf() - dayjs(b.creationDate).valueOf(),
+  );
+
+  return sortedComments.reduce(
+    (acc: CommentOrDivider[], comment: Comment, index: number) => {
+      const currentDivider = dayjs(comment.creationDate).isSame(dayjs(), "day")
+        ? "Aujourd'hui"
```
Comment (niko-amosse) [inline frontend/src/components/comment-panel/utils.ts:20]: i18n ?

File: frontend/src/components/comment-panel/utils.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        : dayjs(comment.creationDate).isSame(dayjs().subtract(1, "day"), "day")
+        ? "Hier"
```
Comment (niko-amosse) [inline frontend/src/components/comment-panel/utils.ts:22]: i18n ?

File: frontend/src/components/comment-panel/utils.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+      const needsDivider =
+        index === 0 ||
+        (() => {
+          const prevComment = sortedComments[index - 1];
+          const prevDivider = dayjs(prevComment.creationDate).isSame(
+            dayjs(),
+            "day",
+          )
+            ? "Aujourd'hui"
```
Comment (niko-amosse) [inline frontend/src/components/comment-panel/utils.ts:33]: i18n ?

**Review by niko-amosse — APPROVED** [review]

File: frontend/src/components/comment-panel/CommentPanel.tsx [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            ),
+          )}
+        </Box>
+        <Box sx={commentPanelFooter}>
+          <Box sx={leftFooterContent}>
+            <Avatar sx={avatarStyle} src={avatar}></Avatar>
+            <StyledTextarea
+              minRows={1}
+              maxRows={4}
+              placeholder={`${t("magneto.add.comment")}...`}
```
[the remark this answers is not in the document]
Comment (FlorentMr) [reply frontend/src/components/comment-panel/CommentPanel.tsx:162]: https://dictionnaire.lerobert.com/guide/points-de-suspension#:~:text=Les%20points%20de%20suspension%20sont,phrase%20qui%20se%20poursuit%20ensuite.

File: frontend/src/components/comment-panel/utils.ts (hunk 3 above)
Comment (FlorentMr) [inline frontend/src/components/comment-panel/utils.ts:20]: Const

File: frontend/src/components/comment-panel/utils.ts (hunk 4 above)
Comment (FlorentMr) [inline frontend/src/components/comment-panel/utils.ts:22]: Const

File: frontend/src/components/comment-panel/utils.ts (hunk 5 above)
Comment (FlorentMr) [inline frontend/src/components/comment-panel/utils.ts:33]: Const

File: frontend/src/components/comment-panel/utils.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            : dayjs(prevComment.creationDate).isSame(
+                dayjs().subtract(1, "day"),
+                "day",
+              )
+            ? "Hier"
```
Comment (FlorentMr) [inline frontend/src/components/comment-panel/utils.ts:38]: Const

**Review by FlorentMr — APPROVED** [review]

File: frontend/src/components/comment-panel-item/style.ts (hunk 2 above)
Comment (alicedraillard) [reply frontend/src/components/comment-panel-item/style.ts:78]: idem

**Review by alicedraillard — APPROVED** [review]

**Pull request merged**